executor/write.go

```diff
@@ -188,3 +188,15 @@ func getTableOffset(schema *expression.Schema, handleCol *expression.Column) int
 	}
 	panic("Couldn't get column information when do update/delete")
 }
+
+func batchDMLCommit(gctx context.Context, ctx sessionctx.Context) error {
+	if err := ctx.StmtCommit(); err != nil {
+		return ErrBatchDMLFail.GenWithStackByArgs(err)
+	}
+	ctx.GetSessionVars().TxnCtx.IsBatched = true
+	if err := ctx.NewTxn(gctx); err != nil {
+		return ErrBatchDMLFail.GenWithStackByArgs(err)
+	}
+	ctx.GetSessionVars().TxnCtx.IsBatched = true
```

Comment:
Repeated? see line 196

Reply:
The `TxnCtx` is refreshed at line 196, so `IsBatched` is set once again.

Reply:
Get it